apps/web-giddh/src/app/invoice/eWayBill/eWayBill/eWayBill.component.ts

```diff
@@ -757,14 +766,51 @@ export class EWayBillComponent implements OnInit, OnDestroy {
      * @memberof EWayBillComponent
      */
     public onTabChange(event: MatTabChangeEvent): void {
-        if (event) {
-            this.activeTabIndex = event.index;
-            this.selectedTab = event.tab.textLabel;
+        if (!event || event.index === this.activeTabIndex) return;
+
+        const colsToRemove = ['status', 'reason', 'ewbNo', 'ewayBillDate'];
+        this.displayedColumns = this.displayedColumns.filter(col => !colsToRemove.includes(col));
+        if (event.index === 0) {
+            this.displayedColumns.splice(-2, 0, 'ewbNo', 'ewayBillDate');
+            this.EwayBillfilterRequest.failedRequestLog = false;
+        } else if (event.index === 1) {
+            this.displayedColumns.splice(-2, 0, 'status', 'reason');
+            this.EwayBillfilterRequest.failedRequestLog = true;
         }
+        this.activeTabIndex = event.index;
+        this.selectedTab = event.tab.textLabel;
+        this.getAllFilteredInvoice();
     }
 
     /**
-     * Handles page change events and makes an API call to fetch data for the new page.
+     * This will generate eway bill for selected voucher
+     *
+     * @param {any} voucher
+     * @memberof EWayBillComponent
+     */
+    public onGenerateEwayBill(voucher: any): void {
+        this.store.dispatch(this.invoiceReceiptActions.ResetVoucherDetails());
+        this._invoiceService.selectedInvoicesLists = [];
```

Comment:
```suggestion
        this.invoiceService.selectedInvoicesLists = [];
```

Please repace in over all file

Reply:
Done